### PR TITLE
London Class 7 - Ionut Tripa - JavaScript Core 1 - Week 3

### PR DESCRIPTION
**Pull request by iTrip23**
Ionut Tripa
London
Johnny

# Homework Details

JavaScript Core 1
Week 3

### CODE REVIEW

File: 2-mandatory/7-password-validator.js [hunk 1]
```diff
@@ -22,9 +22,12 @@ PasswordValidationResult=  [false, false, false, false, true]
 
 */
 
-function validatePasswords(passwords) {
-
-}
+let validatePasswords = passwords => passwords.map((string, index) => /[A-Z]/.test(string)
```
Comment (Laura-TM) [inline 2-mandatory/7-password-validator.js:25]: Love the use of regular expressions here, I need to learn these!

**Comment by Laura-TM** [comment]
Overall, I can only say the way you write code is quite impressive, so compact. Not sure that everyone could read or easily understand it though.

File: 1-exercises/H-array-methods-2/exercise2.js [hunk 2]
```diff
@@ -3,11 +3,11 @@
   -------------------------
   Complete the capitalise function 
   It should return a string with the first letter in uppercase
-  For example, capitailise("hello") should return "Hello"
+  For example, capitalise("hello") should return "Hello"
   Tip: use the string method .split() and the array method .join()
 */
 
-function capitalise(str) {}
+let capitalise = str => str.slice(0, 1).toUpperCase() + str.slice(1).toLowerCase();
```
Comment (omarsharif-gr) [inline 1-exercises/H-array-methods-2/exercise2.js:10]: Looks great - nice alternative! Could you possibly use the .sort() and the .join() methods to achieve the same result here?

File: 2-mandatory/7-password-validator.js [hunk 3]
```diff
@@ -22,9 +22,12 @@ PasswordValidationResult=  [false, false, false, false, true]
...
 
-function validatePasswords(passwords) {
-
-}
+let validatePasswords = passwords => passwords.map((string, index) => /[A-Z]/.test(string) 
+&& /[a-z]/.test(string) 
+&& /[0-9]/.test(string) 
+&& /[!|#|$|%|.|*|&]/.test(string) 
+&& string.length >= 5 
+&& passwords.indexOf(string) === index);
```
Comment (AhmadJanah) [inline 2-mandatory/7-password-validator.js:31]: It is fantastic works, thank you.

I think that it is better to use the functions predefined below this function, but you already did it perfectly.

**Review by AhmadJanah — COMMENTED** [review]
Hi Lonut,

I tried to find any important comment to put it as it is required, but you did everything in a perfect way. Well done. Your code is excellent.

Regards,

File: 1-exercises/H-array-methods-2/exercise2.js (hunk 2 above)
Comment (iTrip23) [reply 1-exercises/H-array-methods-2/exercise2.js:10]: Hi Omar, not sure how can I use sort() in this case.

**Pull request closed without merging**